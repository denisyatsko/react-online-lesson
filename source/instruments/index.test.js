// Core
import { sum, delay, getUniqueID, getFullApiUrl } from './';

describe('instruments:', () => {
    test('sum function should be a function', () => {
        expect(sum).toBeInstanceOf(Function);
    });

    test('sum function should throw, when called with non-number type as second argument', () => {
        expect(() => sum(2, 'привет')).toThrow();
    });

    test('sum function should throw, when called with non-number type as first argument', () => {
        expect(() => sum('привет', 2)).toThrow();
    });

    test('delay function should return an addition of two arguments passed', () => {
        expect(sum(3,3)).toBe(6);
        expect(sum(1,8)).toMatchSnapshot();
    });

    test('delay function should return a resolved promise', async () => {
        await expect(delay()).resolves.toBeUndefined();
    });

    test('getUniqueID function should be a function', () => {
        expect(getUniqueID).toBeInstanceOf(Function);
    });

    test('getUniqueID function should throw, when called with non-number type as an argument', () => {
        expect(() => getUniqueID('привет')).toThrow();
    });

    test('getUniqueID function should produce a string of a desired given length', () => {
        expect(typeof getUniqueID()).toBe('string');
        expect(getUniqueID(5)).toHaveLength(5);
        expect(getUniqueID(12)).toHaveLength(12);
    });

    test('getFullApiUrl function should be a function', () => {
        expect(getFullApiUrl).toBeInstanceOf(Function);
    });

    test('getFullApiUrl function should throw, when called with non-string type as first argument', () => {
        expect(() => getFullApiUrl('привет', 2)).toThrow();
    });

    test('getFullApiUrl function should throw, when called with non-string type as second argument', () => {
        expect(() => getFullApiUrl(2, 'привет')).toThrow();
    });

    test('getFullApiUrl function should return arguments passed should be a string', () => {
        expect(typeof getFullApiUrl('api', 'ID')).toBe('string');
        expect(getFullApiUrl('api', 'ID')).toMatchSnapshot();
    });

});
