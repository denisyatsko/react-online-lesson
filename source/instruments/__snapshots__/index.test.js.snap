// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`instruments: delay function should return an addition of two arguments passed 1`] = `9`;

exports[`instruments: getFullApiUrl function should return arguments passed should be a string 1`] = `"api/ID"`;
